src/Engine/Utils.java

```diff
@@ -175,6 +175,75 @@ public static ArrayList<XYCoord> findFloodFillArea(XYCoord start, Unit mover, Ar
     return reachableTiles;
   }
 
+  public static class PathCalcParams
+  {
+    public XYCoord start;
+    public Unit moverIdentity;
+    public Army team;
+    public MoveType mt;
+    public int initialMovePower;
+    public final GameMap gameMap;
+    public boolean includeOccupiedSpaces;
+    public boolean canTravelThroughEnemies;
+
+    // Does this need a constructor that doesn't require a Unit?
+    public PathCalcParams(Unit unit, GameMap gameMap)
+    {
+      this.gameMap = gameMap;
+      start = new XYCoord(unit);
+      moverIdentity = unit;
+      team = unit.CO.army;
+      mt = unit.getMoveFunctor();
+      initialMovePower = Math.min(unit.getMovePower(gameMap), unit.fuel);
+      includeOccupiedSpaces = true;
+      canTravelThroughEnemies = false;
+    }
+
+    public ArrayList<SearchNode> findAllPaths()
```

Comment:
It looks like a lot of this is copies from `findFloodFillArea` - do you intend to consolidate?

Reply:
Yes, but I was waiting for the "go ahead" on replacing the old API with the new one.

Reply:
Cool. I don't object.